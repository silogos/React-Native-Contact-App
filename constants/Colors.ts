export const Colors = {
  light: {
    primary: '#20A4DC',
    secondary: '#F5F8F9',
    text: '#11181C',
    background: '#fff',
    foreground: '#09090b',
    muted: '#f1f1f1',
    icon: '#687076',
    border: '#e4e4e7',
    card: '#fff',
    input: '#1e293b',
    inputPlaceholder: '#343e4e',
  },
  dark: {
    primary: '#20A4DC',
    secondary: '#F5F8F9',
    text: '#ECEDEE',
    background: '#09090b',
    foreground: '#fafafa',
    muted: '#fafafa',
    icon: '#9BA1A6',
    border: '#e4e4e7',
    card: '#09090b',
    input: '#e2e8f0',
    inputPlaceholder: '#a2a4a6',
  },
};
