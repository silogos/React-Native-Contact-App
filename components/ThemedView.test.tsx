import React from "react";
import { ThemedView } from "./ThemedView";
import { useColorScheme } from "react-native";
import { Colors } from "@/constants/Colors";
import { render } from "@testing-library/react-native";

describe("ThemedView", () => {
  it("renders correctly", () => {
    const elm = render(<ThemedView />).toJSON();

    expect(elm).toMatchSnapshot();
  });

  it("renders correctly with default colorName", () => {
    (useColorScheme as jest.Mock).mockReturnValue("dark");

    const { root } = render(<ThemedView colorName="background" />);

    expect(root.props.style[0].backgroundColor).toEqual(Colors.dark.background);
  });
});
