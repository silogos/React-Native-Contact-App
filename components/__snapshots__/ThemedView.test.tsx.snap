// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
/>
`;
